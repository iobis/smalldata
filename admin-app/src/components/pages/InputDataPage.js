import React from 'react'
import Divider from '../layout/Divider'
import { useTranslation } from 'react-i18next'
import { Link } from 'react-router-dom'

export default function InputDataPage() {
  const { t } = useTranslation()

  return (
    <>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <Link className="button is-info" to="/input-data/new">
            {t('inputDataPage.inputNewOccurrence')}
          </Link>
        </div>
      </section>
      <Divider>{t('inputDataPage.or')}</Divider>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <h4 className="title is-4">{t('inputDataPage.copyPreviousHeader')}</h4>
          <table className="table is-striped is-hoverable is-fullwidth">
            <thead>
              <tr>
                <th/>
                <th>{t('inputDataPage.table.dateAdded')}</th>
                <th>{t('inputDataPage.table.scientificName')}</th>
                <th>{t('inputDataPage.table.dataset')}</th>
                <th>{t('inputDataPage.table.occurrenceDate')}</th>
                <th/>
              </tr>
            </thead>
            <tbody>
            </tbody>
          </table>
        </div>
      </section>
    </>
  )
}
