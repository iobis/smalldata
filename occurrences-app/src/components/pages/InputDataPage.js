import Divider from '../layout/Divider'
import PropTypes from 'prop-types'
import React, { useContext, useEffect, useState } from 'react'
import { AuthContext } from '@smalldata/dwca-lib'
import { format } from 'date-fns'
import { getDatasets, getOccurrences } from '@smalldata/dwca-lib/src/clients/SmalldataClient'
import { Link } from 'react-router-dom'
import { useTranslation } from 'react-i18next'

export default function InputDataPage() {
  const { t } = useTranslation()
  const { userRef } = useContext(AuthContext)
  const [occurrences, setOccurrences] = useState([])

  useEffect(() => {
    const fetchOccurrences = async() => {
      const [occurrences, datasets] = await Promise.all([
        getOccurrences({ userRef }),
        getDatasets()
      ])
      const datasetRefToTitle = datasets
        .reduce((acc, dataset) => {
          acc[dataset.id] = dataset.title.value
          return acc
        }, {})
      const occurrencesWithDataset = occurrences
        .map(occurrence => ({
          ...occurrence,
          datasetTitle: datasetRefToTitle[occurrence.dataset]
        }))
      setOccurrences(occurrencesWithDataset)
    }
    fetchOccurrences()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  return (
    <>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <Link className="button is-info new" to="/input-data/create">
            {t('inputDataPage.inputNewOccurrence')}
          </Link>
        </div>
      </section>
      <Divider>{t('inputDataPage.or')}</Divider>
      <section className="previous-occurrences section">
        <div className="container is-fluid has-text-centered">
          <h4 className="title is-4">{t('inputDataPage.copyPreviousHeader')}</h4>
          <table className="table is-striped is-hoverable is-fullwidth">
            <thead>
              <tr>
                <th/>
                <th>{t('inputDataPage.table.addedAt')}</th>
                <th>{t('inputDataPage.table.scientificName')}</th>
                <th>{t('inputDataPage.table.datasetTitle')}</th>
                <th>{t('inputDataPage.table.occurrenceDate')}</th>
                <th/>
              </tr>
            </thead>
            <tbody>
              {occurrences.map(occurrence => (
                <OccurrenceRow
                  datasetId={occurrence.dataset}
                  key={occurrence.dwcaId}
                  occurrenceDate={occurrence.dwcRecords.occurrence[0].tdwg.eventDate}
                  scientificName={occurrence.dwcRecords.occurrence[0].tdwg.scientificName}
                  {...occurrence}/>
              ))}
            </tbody>
          </table>
        </div>
      </section>
    </>
  )
}

function OccurrenceRow({ addedAtInstant, datasetId, dwcaId, datasetTitle, occurrenceDate, scientificName }) {
  const { t } = useTranslation()
  const addedAtString = addedAtInstant ? format(addedAtInstant, 'D MMMM YYYY') : '—'
  const toCreate = {
    pathname: '/input-data/create',
    state:    { action: 'create', dwcaId, datasetId }
  }
  const toEdit = {
    pathname: '/input-data/update',
    state:    { action: 'update', dwcaId, datasetId }
  }

  return (
    <tr className="occurrence-row">
      <td className="edit">
        <Link className="button is-info" to={toEdit}>
          {t('common.edit')}
        </Link>
      </td>
      <td className="added-at">{addedAtString}</td>
      <td className="scientific-name">{scientificName}</td>
      <td className="dataset-title">{datasetTitle}</td>
      <td className="occurrence-date">{occurrenceDate || '—'}</td>
      <td className="copy">
        <Link className="button is-info copy-previous-entry" to={toCreate}>
          {t('common.copy')}
        </Link>
      </td>
    </tr>
  )
}

OccurrenceRow.propTypes = {
  addedAtInstant: PropTypes.string,
  datasetId:      PropTypes.string.isRequired,
  datasetTitle:   PropTypes.string.isRequired,
  dwcaId:         PropTypes.string.isRequired,
  occurrenceDate: PropTypes.string,
  scientificName: PropTypes.string.isRequired
}
