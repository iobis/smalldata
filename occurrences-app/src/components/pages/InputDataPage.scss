.previous-occurrences {
  margin-top: var(--margin-top);
  .occurrence-row {
    .edit {
      width: 5%;
    }

    .added-at {
      width: 10%;
    }

    .scientific-name {
      width: 10%;
    }

    .dataset-title {
      width: 55%;
    }

    .occurrence-date {
      width: 15%;
    }

    .copy {
      width: 5%;
    }
  }
}

a.button.new-occurrence {
  background-color: var(--obis-green);
  margin-bottom: var(--margin-bottom);

  &:hover {
    background-color: lighten($obis-green, 10%);
  }
}

a.button.is-info.copy-previous-entry {
  background-color: $stepbackground;
}
