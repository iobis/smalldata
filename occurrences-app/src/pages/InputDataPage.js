import React from 'react'
import { getOccurrenceMock } from '../clients/server'
import Divider from '../atom/Divider'
import { useTranslation } from 'react-i18next'

export default function InputDataPage() {
  const { t } = useTranslation()
  const occurrences = getOccurrenceMock()

  return (
    <>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <a className="button is-info">{t('inputData.inputNew')}</a>
        </div>
      </section>
      <Divider>{t('inputData.or')}</Divider>
      <section className="section">
        <div className="container is-fluid has-text-centered">
          <h3>{t('inputData.copyPreviousHeader')}</h3>
          <table className="table is-striped is-hoverable is-fullwidth">
            <thead>
            <tr>
              <th/>
              <th>{t('inputData.table.dateAdded')}</th>
              <th>{t('inputData.table.scientificName')}</th>
              <th>{t('inputData.table.dataset')}</th>
              <th>{t('inputData.table.occurrenceDate')}</th>
              <th/>
            </tr>
            </thead>
            <tbody>
            {occurrences.map(occurrence => <OccurrenceRow key={occurrence.id} {...occurrence}/>)}
            </tbody>
          </table>
        </div>
      </section>
    </>
  )
}

function OccurrenceRow({ id, addedDate, scientificName, dataset, occurrenceDate }) {
  const { t } = useTranslation()

  return (
    <tr>
      <td>
        <div className="button is-info">{t('common.edit')}</div>
      </td>
      <td>{addedDate}</td>
      <td>{scientificName}</td>
      <td>{dataset}</td>
      <td>{occurrenceDate}</td>
      <td>
        <div className="button is-info">{t('common.copy')}</div>
      </td>
    </tr>
  )
}
