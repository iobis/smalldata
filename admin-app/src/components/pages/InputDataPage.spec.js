import InputDataPage from './InputDataPage'
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'

describe('InputDataPage', () => {
  it('renders correctly /', () => {
    expect(renderer.create(<MemoryRouter><InputDataPage/></MemoryRouter>).toJSON()).toMatchSnapshot()
  })
})
