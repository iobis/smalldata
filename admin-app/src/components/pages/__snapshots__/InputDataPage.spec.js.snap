// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDataPage renders correctly / 1`] = `
Array [
  <section
    className="section"
  >
    <div
      className="container is-fluid has-text-centered"
    >
      <a
        className="button is-info"
        href="/input-data/new"
        onClick={[Function]}
      >
        inputDataPage.inputNewOccurrence
      </a>
    </div>
  </section>,
  <div
    className="divider"
  >
    <span>
      inputDataPage.or
    </span>
  </div>,
  <section
    className="section"
  >
    <div
      className="container is-fluid has-text-centered"
    >
      <h4
        className="title is-4"
      >
        inputDataPage.copyPreviousHeader
      </h4>
      <table
        className="table is-striped is-hoverable is-fullwidth"
      >
        <thead>
          <tr>
            <th />
            <th>
              inputDataPage.table.dateAdded
            </th>
            <th>
              inputDataPage.table.scientificName
            </th>
            <th>
              inputDataPage.table.dataset
            </th>
            <th>
              inputDataPage.table.occurrenceDate
            </th>
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </section>,
]
`;
